### PR TITLE
machine_configuration: add MCO to load vfio-pci driver at boot

### PR DESCRIPTION
**Pull request by mregmi**
machine_configuration: Load vfio-pci driver at boot via kernel boot Parameters

Signed-off-by: Manish Regmi <manish.regmi@intel.com>

### CODE REVIEW

**Comment by uMartinXu** [comment]
The patch looks good to me, just need to change the commit log a little bit.
@hershpa please also have a look at it.

**Comment by mythi** [comment]
> 3\. Future Improvement: If possible, can we maintain only 1 QAT machine configuration file that loads `vfio-pci` and sets kernel boot parameter `intel_iommu=on`

I was hoping this PR would already do that because *both* can be set via boot parameters:  `modules_load=vfio-pci vfio-pci.ids=8086:4941,8086:4943`

**Comment by mregmi** [comment]
> > 3. Future Improvement: If possible, can we maintain only 1 QAT machine configuration file that loads `vfio-pci` and sets kernel boot parameter `intel_iommu=on`
> 
> I was hoping this PR would already do that because _both_ can be set via boot parameters: `modules_load=vfio-pci vfio-pci.ids=8086:4941,8086:4943`

Ahh did not know we could do vfio-pci in boot parameter itself. i will make those changes

**Comment by hershpa** [comment]
> > 3. Future Improvement: If possible, can we maintain only 1 QAT machine configuration file that loads `vfio-pci` and sets kernel boot parameter `intel_iommu=on`
> 
> I was hoping this PR would already do that because _both_ can be set via boot parameters: `modules_load=vfio-pci vfio-pci.ids=8086:4941,8086:4943`

Thanks @mythi!

**Comment by mythi** [comment]
> > > 3. Future Improvement: If possible, can we maintain only 1 QAT machine configuration file that loads `vfio-pci` and sets kernel boot parameter `intel_iommu=on`
> > 
> > 
> > I was hoping this PR would already do that because _both_ can be set via boot parameters: `modules_load=vfio-pci vfio-pci.ids=8086:4941,8086:4943`
> 
> Ahh did not know we could do vfio-pci in boot parameter itself. i will make those changes

I hope it also works with OCP.

**Comment by hershpa** [comment]
We still need to test it

File: machine_configuration/100-intel-qat-intel-iommu-on.yaml [hunk 1]
```diff
@@ -12,6 +12,6 @@ spec:
     ignition:
       version: 3.2.0
   kernelArguments:
-      - intel_iommu=on
+      - intel_iommu=on vfio-pci.ids=8086:4941,8086:4943
```
Comment (mythi) [inline machine_configuration/100-intel-qat-intel-iommu-on.yaml:15]: is this enough or `modules_load=vfio-pci` needed too?

Comment (mregmi) [reply machine_configuration/100-intel-qat-intel-iommu-on.yaml]: Yes those were needed. updated the change and verified in our cluster

**Comment by mregmi** [comment]
Tested and verified in our cluster. Please review.

**Comment by hershpa** [comment]
LGTM

**Review by mythi — APPROVED** [review]
nice

**Review by hershpa — APPROVED** [review]

**Comment by uMartinXu** [comment]
it looks good to me.

**Pull request merged**